message-type: Add support for "channel" as value for type parameter.

laurynmm opened the pull request:
For endpoints with a type parameter to indicate whether a message is a direct or stream message, adds support for passing "channel" as a value for stream messages.

Part of stream to channel rename project, see [relevant CZO discussion](https://chat.zulip.org/#narrow/stream/438-release-management/topic/rename.20.22stream.22.20to.20.22channel.22.20project.2C.20.2328468/near/1758864).

**Screenshots and screen captures:**

<details>
<summary>API update</summary>

![Screenshot from 2024-04-10 21-04-34](https://github.com/zulip/zulip/assets/63245456/888db6fa-aa42-4b63-b488-79497c463530)
</details>
<details>
<summary>Typing endpoint</summary>

[Current documentation](https://zulip.com/api/set-typing-status#parameter-type)
![Screenshot from 2024-04-10 21-05-06](https://github.com/zulip/zulip/assets/63245456/e6fb6c48-5484-4844-abd0-20c32700ee3f)
</details>
<details>
<summary>Send message endpoint</summary>

[Current documentation](https://zulip.com/api/send-message#parameter-type)
![Screenshot from 2024-04-10 21-05-18](https://github.com/zulip/zulip/assets/63245456/5dc41e29-32fe-45e0-b1bb-33936c277de5)
</details>
<details>
<summary>Schedule message endpoint</summary>

[Current documentation](https://zulip.com/api/create-scheduled-message#parameter-type)
![Screenshot from 2024-04-10 21-05-37](https://github.com/zulip/zulip/assets/63245456/749fcebc-bb1a-4ce7-8a73-d4fd641f1dc8)
</details>
<details>
<summary>Edit scheduled message endpoint</summary>

[Current documentation](https://zulip.com/api/update-scheduled-message#parameter-type)
![Screenshot from 2024-04-10 21-05-48](https://github.com/zulip/zulip/assets/63245456/b3a2c126-ec9e-4dda-abbe-483fae156db5)
</details>

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

laurynmm commented:
CI failure seems to be [an issue](https://chat.zulip.org/#narrow/stream/43-automated-testing/topic/main.20failing/near/1777765) that's probably not related to these changes.

timabbott commented on zerver/openapi/zulip.yaml:
```diff
@@ -5677,17 +5677,22 @@ paths:
+
+                    **Changes**: In Zulip 9.0 (feature level 248), `"channel"` was added as
+                    an additional value for this parameter to indicate the type of a stream
+                    message.
```
Hmm, should this be above the 7.0 item? I'm not sure why it didn't have a "Changes" heading, but guessing that's a typo, both here and in the other scheduled messages entry.

laurynmm replied:
It wasn't a **Changes** note because these endpoints were added after that feature level. The type was changed in feature level 174 and the endpoint was added in 179.

I think we can drop the first sentence and just leave the bit about "private" being deprecated.

timabbott replied:
OK.

timabbott commented:
This lgtm aside from the two small items noted above.

laurynmm commented:
**Updates**:
- Updated descriptions of scheduled message `type` parameters.
- Added a test to just be clear what happens if the scheduled message `type` is the only edit/change sent.

<details>
<summary>Updated create scheduled message endpoint - type parameter description</summary>

[Current documentation](https://zulip.com/api/create-scheduled-message#parameter-type)
![Screenshot from 2024-04-11 15-11-29](https://github.com/zulip/zulip/assets/63245456/47fc7793-e9ed-486d-ae4b-c7519e537a0b)
</details>
<details>
<summary>Updated edit scheduled message endpoint - type parameter description</summary>

[Current documentation](https://zulip.com/api/update-scheduled-message#parameter-type)
![Screenshot from 2024-04-11 15-10-50](https://github.com/zulip/zulip/assets/63245456/a8e998b7-63eb-4308-90c2-62fdb1365bcd)
</details>

timabbott commented:
Great, merged, thanks @laurynmm! Glad to have another chunk of this migrated.
